### PR TITLE
Mln635 delayed job updates user

### PR DESCRIPTION
**Pull request by daryanypl**

### CODE REVIEW

File: app/jobs/find_available_user_barcode_job.rb
Comment (gonuguntla) [inline app/jobs/find_available_user_barcode_job.rb:5]: @daryanypl  Please can you add readme document for this job.

File: app/models/user.rb
Comment (thisisstephenbetts) [inline app/models/user.rb:503]: Please add into this comment that `pin` is not persisted in the database, so it is not possible to retrieve it later

**Review by thisisstephenbetts — CHANGES_REQUESTED** [review]
Thanks. I've a few comments, but the main issue is the blank pin_code conditional.

File: app/models/user.rb
Comment (daryanypl) [reply app/models/user.rb:511]: This functionality will be changed once we upgrate to 5.2.  For now, this is what we need for this PR.  But stay tuned, for future PRs.

File: app/controllers/registrations_controller.rb
Comment (daryanypl) [reply app/controllers/registrations_controller.rb:51]: I'd rather not do that.  In my experience as a maintainer, I've found that handling each kind of exception separately allows me to include specific information in the logs, to better see code origins in stack traces, to be able to binding.pry and step through the code more smoothly.  This is a personal preference, and I can see how you might prefer to opt for shorter code.  However, personal preference doesn't rise to level of "bug, can damage code base", and I don't believe it should hold up a PR. 
As the project's maintainer, I'd rather keep the code the way I would find it manageable.

File: app/controllers/registrations_controller.rb
Comment (daryanypl) [reply app/controllers/registrations_controller.rb:65]: You're right, it bears a look.  However, I see some Exceptions::InvalidResponse statements in the logs.  I will keep it for now, and add a backlog ticket to look further into this.

File: app/jobs/find_available_user_barcode_job.rb
Comment (daryanypl) [reply app/jobs/find_available_user_barcode_job.rb:98]: I've been able to trigger it once or twice in testing, yes.  Don't remember offhand when.

File: app/models/user.rb
Comment (daryanypl) [reply app/models/user.rb:134]: We have a backlog jira ticket for this discussion.

File: app/jobs/find_available_user_barcode_job.rb
Comment (daryanypl) [reply app/jobs/find_available_user_barcode_job.rb]: done

File: app/models/user.rb
Comment (daryanypl) [reply app/models/user.rb]: done

File: app/jobs/find_available_user_barcode_job.rb
Comment (daryanypl) [reply app/jobs/find_available_user_barcode_job.rb:98]: I made a jira ticket for the backlog, to remind ourselves to re-do exception handling, once we upgrade rails.

**Pull request merged**
